core/src/main/java/org/infinispan/configuration/parsing/Parser80.java

```diff
@@ -1538,7 +1539,15 @@ private void parseInvalidationCache(XMLExtendedStreamReader reader, Configuratio
       for (int i = 0; i < reader.getAttributeCount(); i++) {
          String value = replaceProperties(reader.getAttributeValue(i));
          Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
-         this.parseClusteredCacheAttribute(reader, i, attribute, value, builder, baseCacheMode);
+         switch (attribute) {
+            case KEY_PARTITIONER_FACTORY: {
+               builder.clustering().hash().keyPartitionerFactory(Util.getInstance(value, holder.getClassLoader()));
```

Comment:
As discussed earlier, this PR will only cover library, we need to decide on how to support that since the HotRod client would need to use that same function